✨ Feat: #7 #10 Level 3. 구현

amazingDove367 opened the pull request:
# Level 3: Dedication, Summary 구성
### 구현내용
1. `UIStackView`와 `UILabel`을 사용해서 Dedication과 Summary 구성한다.
2. UI 속성 구현
    - Dedication 속성, Summary 속성

# Level 3: View 파일 분리
1. `HarryPotterSeriseView.swift` 파일 이름 변경 → `HomeView.swift`
    - 더 용도에 맞는 이름으로 변경 
4. Level 2의 책 정보 영역을 `InfoView.swift`로 분리시킨다.
5. Level 3의 Dedication, Summary 영역 → `LableContentView.swift`로 분리시킨다.    
    - 각각의 구성을 해당 `UIView`를 재사용하여 `UI`를 구성한다.

amazingDove367 committed:
✨ Feat: #7 Level 3. Dedication, Summary 구성
1. UIStackView와 UILabel을 사용해서 Dedication과 Summary 구성한다.
2. UI 속성 구현
  - Dedication 속성, Summary 속성

meowbutlerdev commented on HarryPotterSeriesHakyung/HarryPotterSeriesHakyung/View/HarryPotterSeriseView.swift:
```diff
@@ -35,8 +36,15 @@ class HarryPotterSeriseView: UIView {
+    
     private var books: [BookResource] = BookResource.demo()
-    private var dataIndex: Int = 0
+    private var dataIndex: Int = .zero
```
바로 수정하셨네여

amazingDove367 replied:
ㅋㅋㅋ 프리뷰 올렸나요? 진짜 너무 정신없네여 -.- 
바로 적용합니다. ㅋㅋ

meowbutlerdev commented on HarryPotterSeriesHakyung/HarryPotterSeriesHakyung/View/HarryPotterSeriseView.swift:
```diff
@@ -75,10 +83,14 @@ class HarryPotterSeriseView: UIView {
         pagesNumber.text = "\(books[dataIndex].pages)"
+        
+        dedicationContent.text = "\(books[dataIndex].dedication)"
+        summaryContent.text = "\(books[dataIndex].summary)"
```
이 부분은 아마 `pagesNumber.text = "\(books[dataIndex].pages)"` 이걸 복붙 하신거겠죠?
어차피 같은 문자열이라 `"\( )"` 이렇게 감싸지 않아도 될 것 같습니다!

amazingDove367 replied:
바쁘신데 자세히 봐주셔서 감사해요 ㅠㅠ.
아마 저 데이터 왔다갔다 하는거는.. 나중에 손을 또보려고요..
일단 진도 따라잡고..

dawnie-g reviewed (approved):
빠르고 깔끔하게 구현하신 것 같아요! .zero 바로 적용하신것도 봤습니다 ㅋㅋㅋ 고생하셨어요.
후딱 레벨 4 들어갑시다 👏👏

meowbutlerdev commented on HarryPotterSeriesHakyung/HarryPotterSeriesHakyung/View/HarryPotterSeriseView.swift:
```diff
@@ -174,6 +186,38 @@ class HarryPotterSeriseView: UIView {
+        dedicationLabel.font = .systemFont(ofSize: Constants.Text.textSize18, weight: .bold)
+        dedicationLabel.textColor = Constants.Text.textColorDefault
+        // dedicationContent
+        dedicationContent.text = "\(books[dataIndex].dedication)"
```
`dedicationContent.text = "\(books[dataIndex].dedication)"`
`summaryContent.text = "\(books[dataIndex].summary)"`

이 부분들이 `updateUI()`와 `configUI()`에서 중복으로 처리되고 있는데요.
아마 초기화할 때랑 데이터가 바뀌었을 때를 고려해서 작성하신 것 같습니다.

제가 Combine을 잘 몰라서 그러는데,
`updateUI()`에서 다른 설정을 먼저 한 후 실제로 데이터를 받아온 후에
```swift
viewModel.books
    .receive(on: RunLoop.main)
    .sink { [unowned self] books in
        self.dataIndex = Int.random(in: 0..<books.count)
        updateUI(books, dataIndex)
    }.store(in: &subscriptions)
```
이 부분에서 받아온 데이터로 문자열을 넣어주면 안되는 건가요??

amazingDove367 replied:
이거 답하느라 언급해주신것 이것저것 해보다가.. 
지금 약간 뇌에 과부화가 와서 조금 이따가 다시 작성하겠습니다 ㅎ

meowbutlerdev replied:
ㅠㅠ
무시하셔도 됩니다..

우선은 과제 완료에 집중하는걸로 해요.

amazingDove367 replied:
ㅋㅋㅋ 아이거 초기와 중복문제 레벨 2부터 고치고 싶었던거였어요

meowbutlerdev reviewed (approved):
수고하셨습니다!

이번에는 preview도 같이 올려주셔네요 ㅋㅋㅋ
두 분 다 Combine을 쓰시니 저도 써야되나? 고민이 되네요...

amazingDove367 committed:
🎨 UI: #10 Level 3. View 파일 분리
1. HarryPotterSeriseView.swift 파일 이름 변경 → HomeView.swift
   - 더 용도에 맞는 이름으로 변경
2. Level 2의 책 정보 영역을 InfoView.swift로 분리시킨다.
3. Level 3의 Dedication, Summary 영역 → LableContentView.swift로 분리시킨다.
   - 각각의 구성을 해당 UIView를 재사용하여 UI를 구성한다.

amazingDove367 commented:
> 이번에는 preview도 같이 올려주셔네요 ㅋㅋㅋ 두 분 다 Combine을 쓰시니 저도 써야되나? 고민이 되네요...

@meowbutlerdev 
아는게 컴바인 밖에 없어서 그래요 ㅎ
안다고 해봤자 컴바인도 롱타임어고에 공부했던 거라서 기억이 가물가물해서.. 이것도 다시 천천히 써보고 있어요 ㅎ
한번도 써보지 않은 Rx 공부용으로 쓰고 싶은데, 시간이 없어서 아는걸로 일단..
